### PR TITLE
Object oriented parser, header parse capability, header tests

### PR DESCRIPTION
**Pull request by nodatall**

### CODE REVIEW

File: bin/parse [hunk 1]
```diff
@@ -13,7 +14,8 @@ if (!fileName) {
 
 switch (args.paradigm) {
   case 'oo':
-    // load and use object oriented parser
+    const insaneParser = new ooParser( fileContents )
```
Comment (pllearns) [inline bin/parse:17]: Please check on your naming conventions, is this explicit enough of a name for this variable?

File: lib-object-oriented/parser.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+export default class Parser {
+  constructor ( fileContent ) {
+    this.content = fileContent
+    this.tokens = []
+  }
+
+  getTokens(){
+    const lexer = new Lexer( this.content )
+    this.tokens = lexer.lex()
+    }
```
Comment (pllearns) [inline lib-object-oriented/parser.js:13]: Check indentation, should be outdented.

File: lib-object-oriented/tags/p.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default class P {
+  
+}
```
Comment (pllearns) [inline lib-object-oriented/tags/p.js]: Does this need to be added to this commit? Please elaborate if so.

File: lib-object-oriented/token.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+const DOT = '.'
+const HYPHEN = '-'
+const NEW_LINE = 'new_line'
+const NUMB = 'numb'
+const OCTOTHORP = "#"
+const SPACE = ' '
+const TICK = '`'
+const TILDE = '~'
+const UNDERSCORE = '_'
+
```
Comment (pllearns) [inline lib-object-oriented/token.js]: Is this whitespace needed for some reason?

File: spec/heading.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import chai, { expect } from 'chai'
+
+const fs = require('fs')
+const h1 = fs.readFileSync('src/heading.md', 'utf8')
+import ooParser from '../lib-object-oriented/parser'
+
+describe('heading', () => {
+
+  context('object-oriented', () =>{
```
Comment (pllearns) [inline spec/heading.js]: Check space between rocket arrow and curly brace.

File: spec/paragraph.js [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
   context('object-oriented', () =>{
+    const miraculousParser = new ooParser( paragraph )
 
-    // it('wraps single line in <p> tag', () => {
-    //   expect(ooParser.outputMarkdown())
-    //     .to.equal('<p>Crying is beautiful.</p>')
-    // })
+    it('wraps single line in <p> tag', () => {
+    })
   })
```
Comment (pllearns) [inline spec/paragraph.js]: This could be excessive whitespace. Please refactor.

**Review by pllearns — COMMENTED** [review]
Please take a look, great code!

File: bin/parse (hunk 1 above)
Comment (nodatall) [inline bin/parse]: NEVER!

Comment (nodatall) [reply bin/parse:17]: \>_<

**Pull request merged**